### PR TITLE
Odds and Ends: AbsSys, Python3 

### PR DESCRIPTION
**Pull request by profxj**
Series of minor edits to:
  a.  Refactor abs_type in AbsSystem
  b.  Deal with some syntax/bugs when running in Python 3
  c.  Some GUI edits
  d.  Added a test to cover AbsSystem.from_dict and from_json
  e.  Addresses 'crash' when running measure_EW or measure_AODM on a 'bad' spectrum

### CODE REVIEW

File: linetools/isgm/abssystem.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ def from_abslines(cls, abslines, vlim=None, **kwargs):
         return slf
 
     @classmethod
-    def from_components(cls, components, vlim=None):
+    def from_components(cls, components, vlim=None, NHI=None):
         """Instantiate from a list of AbsComponent objects
 
         Parameters
```
Comment (ntejos) [inline linetools/isgm/abssystem.py]: Please add a note about `NHI` parameter in docstring.

**Comment by ntejos** [comment]
Left a minor comment in; all the rest look fine to me.

File: linetools/isgm/abssystem.py (hunk 1 above)
Comment (profxj) [inline linetools/isgm/abssystem.py]: Added

**Comment by profxj** [comment]
Merging..

**Pull request merged**